lms/djangoapps/completion/handlers.py

```diff
@@ -21,9 +23,14 @@ def scorable_block_completion(sender, **kwargs):  # pylint: disable=unused-argum
     """
     if not waffle.waffle().is_enabled(waffle.ENABLE_COMPLETION_TRACKING):
         return
-    user = User.objects.get(id=kwargs['user_id'])
     course_key = CourseKey.from_string(kwargs['course_id'])
     block_key = UsageKey.from_string(kwargs['usage_id'])
+    block_cls = XBlock.load_class(block_key.block_type)
```

Comment:
@robrap @ormsbee The "Resolved" note on the original Author concern isn't as clear as I thought it was when I wrote it.  I meant that I think the problem has been resolved.  

There was a previous implementation that had to call the module store to get the block to see if it was completable, when all I needed was a class attribute.  XBlock.load_class fixes that problem by returning the class object given its string representation from `UsageKey.block_type`.  I think this will be plenty speedy now.  (I'm pretty sure it's little more than a method call and a dict access).

Reply:
LGTM.

Reply:
Thanks for clarifying @jcdyer.  Sounds good.